rapids_build_backend/utils.py

```diff
@@ -9,3 +9,21 @@ def _get_pyproject(dirname: str = ".") -> tomlkit.toml_document.TOMLDocument:
     """Parse and return the pyproject.toml file."""
     with open(os.path.join(dirname, "pyproject.toml")) as f:
         return tomlkit.load(f)
+
+
+def _get_setup_py() -> str:
+    """
+    Returns a string with the contents of setup.py,
+    or empty string if it doesn't exist.
+    """
+    # setuptools.build_meta.get_requires_for_wheel() assumes that "setup.py" is directly
+    # relative to the current working directly, so rapids-build-backend can too.
+    #
+    # ref: https://github.com/pypa/setuptools/blob/f91fa3d9fc7262e0467e4b2f84fe463f8f8d23cf/setuptools/build_meta.py#L304
+    setup_py_file = "setup.py"
```

Comment:
For my understanding... what would the benefit of that be?

This is only used in one place (right here in this function), so putting it in a module-level constant wouldn't provide any consistency benefit. And I can't think of a situation where it'd be necessary to patch it (e.g. in tests).

Reply:
Nothing really. Technically faster, but it's not like it's measurable except in nanobenchmarks, and defining it locally is better for readability. I left the note initially thinking it would get used elsewhere then I realized later that it's not, so ignore this.
```
In [5]: x = 1

In [6]: def g():
   ...:     x = 1
   ...:     return x
   ...: 

In [7]: def f():
   ...:     return x
   ...: 

In [8]: %timeit f()
243 ns _ 0.117 ns per loop (mean _ std. dev. of 7 runs, 1,000,000 loops each)

In [9]: %timeit g()
262 ns _ 1.51 ns per loop (mean _ std. dev. of 7 runs, 1,000,000 loops each)

In [10]: %timeit f()
243 ns _ 0.229 ns per loop (mean _ std. dev. of 7 runs, 1,000,000 loops each)

In [11]: %timeit g()
262 ns _ 0.32 ns per loop (mean _ std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
Ok works for me, thanks for talking through it!